File: chinese-family-tree/frontend/lib/models/marriage.dart
class Marriage {
  final int? id;
  final String uuid;
  final int husbandId;
  final int wifeId;
  final String? marriageDate;
  final String? marriagePlace;
  final String marriageType; // primary, secondary, concubine
  final String? endDate;
  final String? endReason;
  final DateTime? createdAt;

  Marriage({
    this.id,
    required this.uuid,
    required this.husbandId,
    required this.wifeId,
    this.marriageDate,
    this.marriagePlace,
    this.marriageType = 'primary',
    this.endDate,
    this.endReason,
    this.createdAt,
  });

  /// Formatted marriage type in Chinese
  String get typeChinese {
    switch (marriageType) {
      case 'primary':
        return '正室';
      case 'secondary':
        return '侧室';
      case 'concubine':
        return '妾';
      default:
        return '其他';
    }
  }

  factory Marriage.fromJson(Map<String, dynamic> json) {
    return Marriage(
      id: json['id'] as int?,
      uuid: json['uuid'] as String,
      husbandId: json['husband_id'] as int,
      wifeId: json['wife_id'] as int,
      marriageDate: json['marriage_date'] as String?,
      marriagePlace: json['marriage_place'] as String?,
      marriageType: json['marriage_type'] as String? ?? 'primary',
      endDate: json['end_date'] as String?,
      endReason: json['end_reason'] as String?,
      createdAt: json['created_at'] != null
          ? DateTime.parse(json['created_at'] as String)
          : null,
    );
  }

  Map<String, dynamic> toJson() {
    return {
      if (id != null) 'id': id,
      'uuid': uuid,
      'husband_id': husbandId,
      'wife_id': wifeId,
      if (marriageDate != null) 'marriage_date': marriageDate,
      if (marriagePlace != null) 'marriage_place': marriagePlace,
      'marriage_type': marriageType,
      if (endDate != null) 'end_date': endDate,
      if (endReason != null) 'end_reason': endReason,
    };
  }

  Marriage copyWith({
    int? id,
    String? uuid,
    int? husbandId,
    int? wifeId,
    String? marriageDate,
    String? marriagePlace,
    String? marriageType,
    String? endDate,
    String? endReason,
    DateTime? createdAt,
  }) {
    return Marriage(
      id: id ?? this.id,
      uuid: uuid ?? this.uuid,
      husbandId: husbandId ?? this.husbandId,
      wifeId: wifeId ?? this.wifeId,
      marriageDate: marriageDate ?? this.marriageDate,
      marriagePlace: marriagePlace ?? this.marriagePlace,
      marriageType: marriageType ?? this.marriageType,
      endDate: endDate ?? this.endDate,
      endReason: endReason ?? this.endReason,
      createdAt: createdAt ?? this.createdAt,
    );
  }
}

File: chinese-family-tree/backend/internal/models/marriage.go
package models

import "time"

// Marriage represents a marriage relationship
type Marriage struct {
	ID            int64     `db:"id" json:"id"`
	UUID          string    `db:"uuid" json:"uuid"`
	HusbandID     int64     `db:"husband_id" json:"husband_id"`
	WifeID        int64     `db:"wife_id" json:"wife_id"`
	MarriageDate  *string   `db:"marriage_date" json:"marriage_date,omitempty"`
	MarriagePlace *string   `db:"marriage_place" json:"marriage_place,omitempty"`
	MarriageType  string    `db:"marriage_type" json:"marriage_type"` // primary, secondary, concubine
	EndDate       *string   `db:"end_date" json:"end_date,omitempty"`
	EndReason     *string   `db:"end_reason" json:"end_reason,omitempty"`
	CreatedAt     time.Time `db:"created_at" json:"created_at"`
}

// CreateMarriageRequest represents the request to create a marriage
type CreateMarriageRequest struct {
	HusbandID     int64   `json:"husband_id" binding:"required"`
	WifeID        int64   `json:"wife_id" binding:"required"`
	MarriageDate  *string `json:"marriage_date,omitempty"`
	MarriagePlace *string `json:"marriage_place,omitempty"`
	MarriageType  string  `json:"marriage_type" binding:"omitempty,oneof=primary secondary concubine"`
	EndDate       *string `json:"end_date,omitempty"`
	EndReason     *string `json:"end_reason,omitempty"`
}

// UpdateMarriageRequest represents the request to update a marriage
type UpdateMarriageRequest struct {
	HusbandID     int64   `json:"husband_id,omitempty"`
	WifeID        int64   `json:"wife_id,omitempty"`
	MarriageDate  *string `json:"marriage_date,omitempty"`
	MarriagePlace *string `json:"marriage_place,omitempty"`
	MarriageType  string  `json:"marriage_type" binding:"omitempty,oneof=primary secondary concubine"`
	EndDate       *string `json:"end_date,omitempty"`
	EndReason     *string `json:"end_reason,omitempty"`
}

// MarriageDetail includes person details
type MarriageDetail struct {
	Marriage
	Husband Person `json:"husband"`
	Wife    Person `json:"wife"`
}
